_posts/2017-11-30-totw-130.md

```diff
@@ -0,0 +1,185 @@
+---
+title: "Tip of the Week #130: Namespace Naming"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/130
+type: markdown
+order: "130"
+---
+
+Originally posted as totw/130 on 2017-02-17
+
+By Titus Winters (titus@google.com)
+
+*The precision of naming takes away from the uniqueness of seeing* — Pierre
+Bonnard
+
+The earliest commit of the Google C++ Style Guide contains the guidance that
+many people are still using for namespace naming. Roughly, this can be
+summarized as "namespaces are derived from package paths," although there's an
+interesting less-followed option that an entire top-level package
+(search/...) could choose a single flat top-level namespace. Following on
+the heels of Java's package naming requirements, this makes a lot of sense: we
+want to be able to uniquely identify symbols in C++ and we want there to be
+uniqueness and consistency in namespace choice.
+
+Except in actuality, we don't. We just didn't realize for almost a decade.
+
+## Name lookup
+
+Let's start with how name lookup works in C++ and how it's different from Java.
+
+``` c++
+namespace foo {
+namespace bar {
+void f() {
```

Comment:
Mildly prefer a blank line between namespace and f(). Don't know if this is a style violation :)

Reply:
yeah, I think that'd be a violation

Reply:
:(